Source/MQTTnet/Server/MqttClientConnection.cs

```diff
@@ -159,13 +167,33 @@ private async Task<MqttClientDisconnectType> RunInternalAsync(MqttConnectionVali
                     Interlocked.Increment(ref _sentPacketsCount);
                     _lastPacketReceivedTimestamp = DateTime.UtcNow;
 
+                    if (packet is MqttAuthPacket clientAuthPacket)
+                    {
+                        if (_serverOptions.EnhancedAuthenticationBrokerHandler == null)
+                        {
+                            Session.WillMessage = null;
+                            disconnectType = MqttClientDisconnectType.NotClean;
+                            await StopAsync();
+                            break;
+                        }
+
+                        var responsePacket = _serverOptions.EnhancedAuthenticationBrokerHandler.HandleAuth(clientAuthPacket, Session.Items);
+                        if (authPacket is MqttConnAckPacket connAckPacket)
+                        {
+                            connAckPacket.IsSessionPresent = !Session.IsCleanSession;
+                        }
```

Comment:
I don't know if I am supposed to deal with this or if it should be here.  If I return a connAckPacket with a failed reason code then the session needs to be shutdown.

Reply:
Similar question arises for a re-authentication where one returns a DISCONNECT packet in v5 and just disconnects the session in v3.